workbench-core/authorization/src/dynamicAuthorization/dynamicAuthorizationInputs/dynamicOperation.ts

```diff
@@ -3,26 +3,31 @@
  *  SPDX-License-Identifier: Apache-2.0
  */
 
-import { Action } from '../../action';
+import { z } from 'zod';
+import { ActionParser } from '../../action';
 
-/**
- * The dynamic operation a {@link User} wants to perform
- */
-export interface DynamicOperation {
+// eslint-disable-next-line @rushstack/typedef-var
+export const DynamicOperationParser = z.object({
   /**
    * The {@link Action} a {@link User} wants to perform.
    */
-  action: Action;
+  action: ActionParser,
   /**
    * The subject that the {@link Action} acts on.
    */
-  subject: {
-    [key: string]: string;
-    subjectType: string;
-    subjectId: string;
-  };
+  subject: z.record(z.string()).and(
```

Comment:
why do we have a union here? Shouldn't everything come with a type and id?

Reply:
you can also add additional attributes about subject here, this can also be extracted into a separate subject attributes section